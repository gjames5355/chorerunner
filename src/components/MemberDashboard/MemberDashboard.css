.memberDashboard {
  position: relative;
  height: 70vh;
  padding: 15vh 0;
  overflow-y: auto;
  background-image: url('../../ImgAssets/ChoreRunner.png');
  background-size: cover;
  background-position: 0 0;
  background-repeat: repeat;
  animation: house 45s linear infinite;
}

.chores-container {
  background: rgba(0, 0, 0, 0.7);
  border-radius: 10px;
  width: 95%;
  margin: auto;
  padding-bottom: 20px;
}

.points span {
  color: white;
  font-size: 18px;
  font-family: 'Electrolize', sans-serif;
}

.taskName p {
  color: white;
  font-size: 30px;
}


.chores-container h2 {
  padding-top: 20px;
  color: rgb(255, 217, 0);
  font-size: 20px;
  text-shadow: none;
  font-family: 'Press Start 2P', cursive;
}

.member-container {
  position: absolute;
  width: 100%;
  padding-bottom: 15vh;
}

.badge-task-container {
  margin: 0 auto 50px auto;
}

.task-control button {
  text-align: center;
  margin: 20px 5px;
  font-family: 'Press Start 2P', cursive;
  font-size: 13px;
  padding: 10px;
  background-color: rgb(255, 217, 0);
  text-shadow: -.5px -1px rgb(105, 90, 0), 1px 1px white;
  color: #725F00;
  -webkit-border-radius: 7px;
  -moz-border-radius: 7px;
  -o-border-radius: 7px;
  border-radius: 7px;
  box-shadow: 0 .3em rgb(134, 112, 0);
  cursor: pointer;
  animation: bounce 500ms infinite alternate;
  -webkit-animation: bounce 500ms infinite alternate;
  cursor: pointer;

}

@keyframes house {
  from {
    background-position: 0 0;
  }

  to {
    background-position: -13690px 0;
  }
}

/* .background-img{
    background-image: url("../../images/ChoreRunner.png");
    background-size: cover;
    height: 100vh;
} */

.task_list ul {
  padding: 0;
}

.task_list li {
  list-style: none;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.task_list p {
  max-width: 200px;
}

.task_list li button {
  margin: 15px auto;
  font-size: 17px;
  min-width: 80px;
}

@media screen and (min-width: 470px) {

  /* 
  .chores-container p,
  .chores-container span {
    color: white;
    font-size: 30px;
  } */
  .chores-container h2 {
    padding-top: 20px;
    color: rgb(255, 217, 0);
    font-size: 20px;
    text-shadow: none;
    font-family: 'Press Start 2P', cursive;
  }

  .member-container {
    position: absolute;
    width: 100%;
  }

  @keyframes house {
    from {
      background-position: 0 0;
    }

    to {
      background-position: -13690px 0;
    }
  }

  /* .background-img{
        background-image: url("../../images/ChoreRunner.png");
        background-size: cover;
        height: 100vh;
    } */
  .task_list li {
    width: 90%;
    margin: 0 auto;
    list-style: none;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
  }

  .task_list p {
    text-align: left;
    max-width: 200px;
  }

  .points {
    /* margin: 0 20px; */
    min-width: 80px;
  }

  .taskName {
    min-width: 45%;
  }

  .task-name {
    min-width: 250px;
  }

}

@media screen and (min-width: 600px) {
  .memberDashboard {
    position: relative;
    height: 70vh;
    overflow-y: auto;
    background-image: url("../../ImgAssets/ChoreRunner.png");
    background-size: cover;
    background-position: 0 0;
    background-repeat: repeat;
    animation: house 45s linear infinite;
  }

  .chores-container {
    background: rgba(0, 0, 0, 0.7);
    border-radius: 10px;
    width: 80%;
    margin: auto;
  }



}

@media screen and (min-width: 850px) {
  .chores-container {
    width: 60%;
  }

  .badge-task-container {
    display: flex;
    flex-direction: column;
    width: 80%;
    margin: 0 auto 100px auto;
    justify-content: space-between;

  }
}

@media screen and (min-width: 1050px) {
  .chores-container {
    width: 700px;
  }
}