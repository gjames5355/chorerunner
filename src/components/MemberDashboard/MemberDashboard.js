import React, { Component } from 'react';
import HouseholdContext from '../../contexts/HouseHoldContext';
import ApiService from '../../services/api-service.js';
import './MemberDashboard.css';
import Leaderboard from '../LeaderBoard/LeaderBoard';
import Badge from '../Badge/Badge';

export default class MemberDashboard extends Component {
  static contextType = HouseholdContext;

  componentDidMount() {
    ApiService.getMemberTasks()
      .then(res => {
        this.context.setMemberTasks(res);
      })
      .catch(error => this.context.setError(error));
  }

  handleCompleted(id) {
    ApiService.completeTask(id)
      .then(this.context.completeTask(id))
      .catch(error => this.context.setError(error));
  }

  renderTasks() {
    const tasks = this.context.memberTasks;
    return tasks.map(task => {
      return (
        <li key={task.id}>
          <div className='taskName'>
            <p>{task.title}</p>
          </div>
          <div className='points'>
            <span>{task.points} EXP</span>
          </div>
          <div className='task-control'>
            <button
              onClick={() => {
                this.handleCompleted(task.id);
              }}
            >
              Clear!
            </button>
          </div>
        </li>
      );
    });
  }

  render() {
    return (
      <section className='memberDashboard'>
        <div className='member-container'>
          <div className='leaderboard_container'>
            <Leaderboard />
          </div>
          <Badge />
          <div className='chores-container'>
            <h2>Chore-llenges</h2>

            <div className='task_list'>
              <ul>{this.renderTasks()}</ul>
            </div>
          </div>
        </div>
      </section>
    );
  }
}
