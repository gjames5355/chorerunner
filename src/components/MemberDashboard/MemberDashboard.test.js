import React from 'react';
import ReactDOM from 'react-dom';
import MemberDashboard from './MemberDashboard';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('MemberDashboard component testing', () => {
  const memberTasks = [
    {
      id: 1,
      title: 'hi',
      points: 1,
    },
    {
      id: 2,
      title: 'hello',
      points: 2,
    },
  ];

  // beforeEach(() => {
  //   memberTasks = shallow(<MemberDashboard tasks={tasks}/>)
  // })

  // const component = shallow(<MemberDashboard memberTasks={memberTasks}/>)

  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <MemoryRouter>
        <MemberDashboard memberTasks={memberTasks} />
      </MemoryRouter>,
      div
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <MemberDashboard memberTasks={memberTasks} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
