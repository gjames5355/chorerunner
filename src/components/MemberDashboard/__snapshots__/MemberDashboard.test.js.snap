// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberDashboard component testing renders UI as expected 1`] = `
<section
  className="memberDashboard"
>
  <div
    className="member-container"
  >
    <div
      className="leaderboard_container"
    >
      <section
        className="leader_board"
      >
        <h3>
          Leaderboard
        </h3>
        <div
          className="leader_board-grid"
        >
          <div
            className="header_rank"
          >
            <span>
              Rank
            </span>
          </div>
          <div
            className="header_name"
          >
            <span>
              Member Name
            </span>
          </div>
          <div
            className="header_score"
          >
            <span>
              Score
            </span>
          </div>
        </div>
        <ul
          className="rankings"
        />
      </section>
    </div>
    <p>
       LOADING... 
    </p>
    <div
      className="chores-container"
    >
      <h2>
        Chore-llenges
      </h2>
      <div
        className="task_list"
      >
        <ul />
      </div>
    </div>
  </div>
</section>
`;
